### PR TITLE
XML Var. substitution - Escape XML characters

### PR DESCRIPTION
**Pull request by vincentdass**
#6814

### CODE REVIEW

File: Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts [hunk 1]
```diff
@@ -262,4 +265,18 @@ function updateXmlConnectionStringsNodeAttribute(xmlDomNode, variableMap, replac
     }
 
     return isSubstitutionApplied;
+}
+
+function replaceEscapeXMLCharacters(xmlDOM) {
```
Comment (Ajay-MS) [inline Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts]: Should we rename xmlDOM to xmlDomNode in method argument?

File: Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts [hunk 2]
```diff
@@ -262,4 +265,18 @@ function updateXmlConnectionStringsNodeAttribute(xmlDomNode, variableMap, replac
...
+    if(!xmlDOM || typeof xmlDOM == 'string') {
+        return;
+    }
+
+    for(var xmlAttributes in xmlDOM.attrs) {
```
Comment (Ajay-MS) [inline Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts]: xmlAttributes  should be singular.

Comment (vincentdass) [reply Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts]: done. thanks.

File: Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts (hunk 1 above)
Comment (vincentdass) [reply Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts]: renamed.

**Pull request merged**